examples/angular/src/pages/ui/components/authenticator/sign-up-with-email/sign-up-with-email.component.ts

```diff
@@ -10,4 +10,20 @@ export class SignUpWithEmailComponent {
   constructor() {
     Amplify.configure(awsExports);
   }
+
+  services = {
+    async handleSignUp(formData: Record<string, any>) {
+      const { username, password, attributes } = formData;
+      // custom username
+      const newAttributes = {
+        ...attributes,
+        email: attributes.email.toLowerCase(),
+      };
+      return Auth.signUp({
+        username: username.toLowerCase(),
+        password,
+        attributes: newAttributes,
+      });
```

Comment:
Would mutations make this example seem a bit clearer (mainly so that that `Auth.signUp` signature is straightforward)?

```suggestion
      let { username, password, attributes } = formData;
      
      username = username.toLowerCase();
      attributes.email = attributes.email.toLowerCase()
      
      return Auth.signUp({ username, password, attributes });
```

Or would something like this pose a risk because we're not explicitly immutable within the machine?

Reply:
Yeah, I could make this change.